app/src/main/java/com/ftw/hometerview/ui/withdrawal/WithdrawalActivity.kt

```diff
@@ -0,0 +1,104 @@
+package com.ftw.hometerview.ui.withdrawal
+
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.ftw.hometerview.R
+import com.ftw.hometerview.databinding.ActivityWithdrawalBinding
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class WithdrawalActivity : AppCompatActivity() {
+
+    companion object {
+        fun newIntent(context: Context): Intent {
+            return Intent(context, WithdrawalActivity::class.java)
+        }
+    }
+
+    @Inject
+    lateinit var viewModel: WithdrawalViewModel
+
+    private lateinit var binding: ActivityWithdrawalBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView<ActivityWithdrawalBinding>(
+            this,
+            R.layout.activity_withdrawal
+        ).apply {
+            viewModel = this@WithdrawalActivity.viewModel
+        }
+        observe()
+    }
+
+    private fun observe() {
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.event.collect { event ->
+                    when (event) {
+                        WithdrawalViewModel.Event.None -> {}
+                        WithdrawalViewModel.Event.OnClickWithdrawalFirstCheck -> onClickWithdrawalFirstCheck()
+                        WithdrawalViewModel.Event.OnClickWithdrawalSecondCheck -> onClickWithdrawalSecondCheck()
+                        WithdrawalViewModel.Event.OnClickWithdrawalThirdCheck -> onClickWithdrawalThirdCheck()
+                    }
+                }
+            }
+        }
+
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.withdrawalCheck.collect { showBanner ->
+                    binding.withdrawalButton.isEnabled = showBanner
+                }
+            }
+        }
+    }
+
+    private fun onClickWithdrawalFirstCheck() {
+        if(viewModel.withdrawalFirstCheck.value){
+            viewModel.withdrawalFirstCheck.value = false
+            binding.withdrawalFirstCheck.setImageResource(R.drawable.icon_check_disabled)
+        } else{
+            viewModel.withdrawalFirstCheck.value = true
+            binding.withdrawalFirstCheck.setImageResource(R.drawable.icon_check_enabled)
+        }
+        viewModel.withdrawalCheck.value = withdrawalCheck()
+    }
```

Comment:
근데 요 로직은 ViewModel 에서 구현할 수 있는 것으로 보이는데, 외부로 나온 이유가 있는거야~?
checkBox 도 selector 로 구현하면 될 것 같은데, drawable 를 따로 변경해주는 이유가 있을까?

아래는 띄어쓰기 부탁합니당!
```suggestion
    private fun onClickWithdrawalFirstCheck() {
        if (viewModel.withdrawalFirstCheck.value) {
            viewModel.withdrawalFirstCheck.value = false
            binding.withdrawalFirstCheck.setImageResource(R.drawable.icon_check_disabled)
        } else {
            viewModel.withdrawalFirstCheck.value = true
            binding.withdrawalFirstCheck.setImageResource(R.drawable.icon_check_enabled)
        }
        viewModel.withdrawalCheck.value = withdrawalCheck()
    }
```

Reply:
ui의 이미지를 바꿔주면서 나오는게 맞다고 생각했었는데, 체크박스 쓰고, selector 써주면 또 바뀔 것 같아요! 바꿔볼게요!